src/Uno.UI.Composition/Composition/ContainerVisual.skia.cs

```diff
@@ -1,10 +1,40 @@
 #nullable enable
 
-using SkiaSharp;
+using System.Collections.Generic;
+using System.Linq;
 
-namespace Windows.UI.Composition
+namespace Windows.UI.Composition;
+
+public partial class ContainerVisual : Visual
 {
-	public partial class ContainerVisual : Visual
+	private List<Visual> _childrenInRenderOrder = new List<Visual>();
+	private bool _hasCustomRenderOrder = false;
+
+	internal bool IsChildrenRenderOrderDirty { get; set; }
+
+	internal IEnumerable<Visual> GetChildrenInRenderOrder()
```

Comment:
We should return the List<T> directly for allocation-less enumeration.

Reply:
Switched to `IList<Visual>` to accomodate for `VisualCollection` and now it should be allocation-free.